imjadn-v1.0-cn02.md

```diff
@@ -2488,28 +2488,38 @@ regarding JADN.
 
 ## D.1 JADN vs. UML Primitive Data Types
 
-[[UML](#uml)] section 21 says "The PrimitiveTypes package is an
-independent package that defines a set of reusable PrimitiveTypes
-that are commonly used in the definition of metamodels."  JADN
-defines an additional Binary type (a sequence of octets/bytes)
-because it is needed. Unlike UML, JADN does not define a separate
-type for UnlimitedNatural because the Integer type can be given
-upper and lower bounds, and natural numbers are the set of
-non-negative integers.  The equivalent in JADN uses
-Integer{0..\*} for natural numbers, and the Integer value -1 for
-the "unlimited" value (*) of UnlimitedNatural. Table D-1 maps
-basic data types between UML and JADN.
+The Universal Modeling Language [UML](#uml) says in section 21:
+"*The PrimitiveTypes package is an independent package that defines a set of
+reusable PrimitiveTypes that are commonly used in the definition of metamodels.*"
 
-###### Table D-1 -- UML and JADN Basic Type Equivalence
+JADN uses the same UML primitive types with two exceptions:
+* JADN defines a Binary (byte sequence) primitive type, UML does not.
+* UML defines an UnlimitedNatural primitive type, JADN does not.
+
+Natural numbers (integers greater than zero) are a subset of integers,
```

Comment:
UML Spec in 21.2 says "An instance of UnlimitedNatural is a value in the (infinite) set of natural numbers (0, 1, 2…) plus
unlimited." So I believe line 2499 should say "(integers greater than _or equal to_ zero)".

Reply:
[Britannica](https://www.britannica.com/science/natural-number) says "natural number, any number in the set of positive integers{1, 2, 3…} and sometimes zero. The term natural likely refers to humanity’s innate understanding and use of positive discrete values. Sometimes called “counting numbers,” natural numbers predate recorded history, as it is believed that prehistoric humans had some sense of determining differences in quantities. Whether zero is considered a natural number differs across definitions, and there is no general consensus on its inclusion or exclusion from the set.

[Carnegie Mellon](http://mathfoundations.lti.cs.cmu.edu/class2/naturals.html) offers an interesting tidbit: "At least some of the controversy regarding 0 is due to the fact that the number 0 was a relatively late entrant to the pantheon of numbers, and is, in some sense, not natural (one elephant differs from one cat, but how do zero elephants differ from zero cats?).

[Many other sources](https://arbs.nzcer.org.nz/types-numbers), though none authoritative for the world of mathematics, declare whole numbers and natural numbers to be different.

But UML says what it says, so I'll make the changes.

Reply:
I have a very good mathematician friend who’s an administrator / professor at Hopkins, author of several books on mathematics, if you want another opinion.  😊

But if we’re comparing JADN to UML I guess we at least need to represent UML as it is.